File: mobile/Application/lib/models/notice.dart
/// 通知公告
class Notice {
  Notice({
    required this.id,
    required this.title,
    required this.content,
    required this.publisher,
    required this.publishTime,
    this.isRead = false,
    this.isImportant = false,
    this.orgId = '',
    this.code = '',
    this.status = 'active',
    this.createdAt,
    this.createdBy = '',
    this.updatedBy = '',
    this.version = 1,
    this.sourceType = 'manual',
    this.sourceId = '',
    this.updatedAt,
  });

  final String id;
  final String title;
  final String content;
  final String publisher;
  final DateTime publishTime;
  bool isRead;
  final bool isImportant;
  String orgId;
  final String code;
  final String status;
  final DateTime? createdAt;
  final String createdBy;
  final String updatedBy;
  final int version;
  final String sourceType;
  final String sourceId;
  final DateTime? updatedAt;

  Map<String, dynamic> toJson() => {
        'id': id,
        'title': title,
        'content': content,
        'publisher': publisher,
        'publishTime': publishTime.millisecondsSinceEpoch,
        'isRead': isRead,
        'isImportant': isImportant,
        'orgId': orgId,
        'code': code,
        'status': status,
        if (createdAt != null) 'createdAt': createdAt!.millisecondsSinceEpoch,
        'createdBy': createdBy,
        'updatedBy': updatedBy,
        'version': version,
        'sourceType': sourceType,
        'sourceId': sourceId,
        if (updatedAt != null) 'updatedAt': updatedAt!.millisecondsSinceEpoch,
      };

  factory Notice.fromJson(Map<String, dynamic> json) => Notice(
        id: json['id'] as String,
        title: json['title'] as String,
        content: (json['content'] as String?) ?? '',
        publisher: (json['publisher'] as String?) ?? '社长办公室',
        publishTime: _toDate(json['publishTime']) ?? DateTime.now(),
        isRead: (json['isRead'] as bool?) ?? false,
        isImportant: (json['isImportant'] as bool?) ?? false,
        orgId: (json['orgId'] as String?) ?? '',
        code: (json['code'] as String?) ?? '',
        status: (json['status'] as String?) ?? 'active',
        createdAt: _toDate(json['createdAt']),
        createdBy: (json['createdBy'] as String?) ?? '',
        updatedBy: (json['updatedBy'] as String?) ?? '',
        version: (json['version'] as num?)?.toInt() ?? 1,
        sourceType: (json['sourceType'] as String?) ?? 'manual',
        sourceId: (json['sourceId'] as String?) ?? '',
        updatedAt: _toDate(json['updatedAt']),
      );
}

/// 兼容 int（epoch 毫秒）/ String（ISO 或数字）两种云端日期形态。
DateTime? _toDate(dynamic v) {
  if (v == null) return null;
  if (v is DateTime) return v;
  if (v is int) return DateTime.fromMillisecondsSinceEpoch(v);
  if (v is String) {
    final n = int.tryParse(v);
    if (n != null) return DateTime.fromMillisecondsSinceEpoch(n);
    return DateTime.tryParse(v);
  }
  return null;
}

File: mobile/CloudProgram/cloudfunctions/upsert-notice/Notice.ts
export class Notice {
  id: string = '';
  title: string = '';
  content: string = '';
  publisher: string = '';
  publishTime: Date = new Date();
  isRead: boolean = false;
  isImportant: boolean = false;
  orgId: string = '';
  code: string = '';
  status: string = 'active';
  createdAt: Date | null = null;
  createdBy: string = '';
  updatedBy: string = '';
  version: number = 1;
  sourceType: string = 'manual';
  sourceId: string = '';
  updatedAt: Date | null = null;

  getClassName(): string {
    return 'Notice';
  }

  getFieldTypeMap(): Map<string, string> {
    const map = new Map<string, string>();
    map.set('id', 'String');
    map.set('title', 'String');
    map.set('content', 'String');
    map.set('publisher', 'String');
    map.set('publishTime', 'Date');
    map.set('isRead', 'Boolean');
    map.set('isImportant', 'Boolean');
    map.set('orgId', 'String');
    map.set('code', 'String');
    map.set('status', 'String');
    map.set('createdAt', 'Date');
    map.set('createdBy', 'String');
    map.set('updatedBy', 'String');
    map.set('version', 'Integer');
    map.set('sourceType', 'String');
    map.set('sourceId', 'String');
    map.set('updatedAt', 'Date');
    return map;
  }

  getPrimaryKeyList(): string[] {
    return ['id'];
  }

  getIndexList(): string[] {
    return ['id'];
  }

  getEncryptedFieldList(): string[] {
    return [];
  }

  static parseFrom(data: any): Notice {
    const obj = new Notice();
    if (data) {
      obj.id = data.id ?? '';
      obj.title = data.title ?? '';
      obj.content = data.content ?? '';
      obj.publisher = data.publisher ?? '';
      obj.publishTime = data.publishTime ? new Date(data.publishTime) : new Date();
      obj.isRead = data.isRead ?? false;
      obj.isImportant = data.isImportant ?? false;
      obj.orgId = data.orgId ?? '';
      obj.code = data.code ?? '';
      obj.status = data.status ?? 'active';
      obj.createdAt = data.createdAt ? new Date(data.createdAt) : null;
      obj.createdBy = data.createdBy ?? '';
      obj.updatedBy = data.updatedBy ?? '';
      obj.version = data.version ?? 1;
      obj.sourceType = data.sourceType ?? 'manual';
      obj.sourceId = data.sourceId ?? '';
      obj.updatedAt = data.updatedAt ? new Date(data.updatedAt) : null;
    }
    return obj;
  }
}
